torch/csrc/distributed/c10d/ProcessGroupNCCL.cpp

```diff
@@ -2428,7 +2430,7 @@ c10::intrusive_ptr<ProcessGroupNCCL::WorkNCCL> ProcessGroupNCCL::initWork(
       device,
       rank,
       opType,
-      seqCollective_,
+      isP2POp(opType) ? seqP2P_ : seqCollective_,
```

Comment:
according to [isP2POp](https://github.com/pytorch/pytorch/blob/1011e0ae9804baca9ef06abfeb6903f15459d346/torch/csrc/distributed/c10d/ProcessGroup.cpp#L65), batched P2P is not P2P op. So we need to check coalesing here.

Reply:
batched P2P is implemented through Coalescing. The id we want to pick is based on if it is either a single P2P op/coalesced P2P op or a single collective op/coalesced collective op, does not matter it is coalesced or not.

Reply:
Ok. If it's coalesed p2p, `seqCollective_` is bumped up in `startCoalesing()` and we pick `seqCollective_` here.
Not sure whether this will be changed/rectified according to this [comment](https://github.com/pytorch/pytorch/issues/134833#issuecomment-2322735566) from @fduwjj